File: fsharp/darts/Darts.fs
module Darts

let score (x: double) (y: double): int =
    let dist = x * x + y * y |> sqrt
    match dist with
    | z when z <= 1.0 -> 10
    | z when z <= 5.0 -> 5
    | z when z <= 10.0 -> 1
    | _ -> 0

File: ruby/darts/darts.rb
class Darts
    def initialize(x, y)
        @@x = x
        @@y = y
    end

    def score
        case dist.ceil
        when 0..1 then 10
        when 2..5 then 5
        when 6..10 then 1
        else 0
        end
    end

    def dist
        Math.sqrt(@@x ** 2 + @@y ** 2)
    end
end